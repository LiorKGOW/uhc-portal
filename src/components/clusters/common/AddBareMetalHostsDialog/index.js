import { connect } from 'react-redux';
import { AddBareMetalHostsDialog } from 'facet-lib';

import { closeModal } from '../../../common/Modal/ModalActions';
import shouldShowModal from '../../../common/Modal/ModalSelectors';
import withFeatureGate from '../../../features/with-feature-gate';
import { ASSISTED_INSTALLER_FEATURE } from '../../../../redux/constants/featureConstants';

const mapStateToProps = (state) => {
  const cluster = state.modal.data;
  return ({
    isOpen: shouldShowModal(state, 'add-bare-metal-hosts'),
    cluster,
  });
};

const mapDispatchToProps = dispatch => ({
  closeModal: () => dispatch(closeModal()),
});

export default withFeatureGate(
  connect(mapStateToProps, mapDispatchToProps)(AddBareMetalHostsDialog),
  ASSISTED_INSTALLER_FEATURE,
  () => null,
);
