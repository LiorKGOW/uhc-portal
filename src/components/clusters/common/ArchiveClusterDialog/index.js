import { connect } from 'react-redux';

import {
  clearClusterArchiveResponse,
  archiveCluster,
} from '../../../../redux/actions/clustersActions';
import ArchiveClusterDialog from './ArchiveClusterDialog';
import { closeModal } from '../../../common/Modal/ModalActions';

const mapStateToProps = (state) => {
  const modalData = state.modal.data;
  return ({
    archiveClusterResponse: state.clusters.archivedCluster,
    subscriptionID: modalData.subscriptionID ? modalData.subscriptionID : '',
    name: modalData.name ? modalData.name : '',
  });
};

const mapDispatchToProps = dispatch => ({
  submit: (subscriptionID, name) => {
    dispatch(archiveCluster(subscriptionID, name));
  },
  resetResponse: () => dispatch(clearClusterArchiveResponse()),
  closeModal: () => dispatch(closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ArchiveClusterDialog);
