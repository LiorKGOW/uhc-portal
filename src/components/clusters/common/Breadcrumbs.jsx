import React from 'react';
import PropTypes from 'prop-types';
import { LinkContainer } from 'react-router-bootstrap';
import {
  Breadcrumb, BreadcrumbItem,
} from '@patternfly/react-core';
import getBaseName from '../../../common/getBaseName';

function Breadcrumbs({ path }) {
  return (
    <Breadcrumb>
      {path.map((item, i) => {
        const itemLabel = item.label;

        if (i < path.length - 1) {
          const itemPath = item.path;

          return (
            <LinkContainer key={itemLabel} to={itemPath || ''}>
              <BreadcrumbItem to={`${getBaseName()}${itemPath || ''}` || '#'}>
                {itemLabel}
              </BreadcrumbItem>
            </LinkContainer>
          );
        }
        return (
          <BreadcrumbItem key={itemLabel} isActive>
            {itemLabel}
          </BreadcrumbItem>
        );
      })}
    </Breadcrumb>
  );
}

Breadcrumbs.propTypes = {
  path: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      path: PropTypes.string,
    }).isRequired,
  ).isRequired,
};

export default Breadcrumbs;
