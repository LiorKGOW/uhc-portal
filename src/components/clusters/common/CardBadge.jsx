import React from 'react';
import PropTypes from 'prop-types';
import { StarIcon } from '@patternfly/react-icons';
// eslint-disable-next-line camelcase
import { global_active_color_100 } from '@patternfly/react-tokens';
import { Label } from '@patternfly/react-core';

function CardBadge({
  isRecommened, isDevPreview, isHidden, devPreviewText = 'Developer Preview',
}) {
  const className = isHidden ? 'card-badge card-badge-hidden' : 'card-badge';
  return (
    <div className="card-badge-container">
      <span className={className}>
        {isRecommened && <StarIcon color={global_active_color_100.value} />}
        {isRecommened && 'Recommended'}
        {isDevPreview && <Label variant="outline" color="orange">{devPreviewText}</Label>}
      </span>
    </div>
  );
}

CardBadge.propTypes = {
  isRecommened: PropTypes.bool,
  isDevPreview: PropTypes.bool,
  isHidden: PropTypes.bool,
  devPreviewText: PropTypes.string,
};

export default CardBadge;
