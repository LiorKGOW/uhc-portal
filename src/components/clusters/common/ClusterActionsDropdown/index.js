import { connect } from 'react-redux';

import { openModal } from '../../../common/Modal/ModalActions';

import ClusterActionsDropdown from './ClusterActionsDropdown';

const mapDispatchToProps = {
  openModal,
};

export default connect(null, mapDispatchToProps)(ClusterActionsDropdown);
