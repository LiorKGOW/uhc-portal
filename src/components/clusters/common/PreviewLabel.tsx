import React from 'react';
import { Label, LabelProps } from '@patternfly/react-core';
import InfoCircleIcon from '@patternfly/react-icons/dist/esm/icons/info-circle-icon';
import { useGetTechPreviewStatus, useTechPreviewStatus } from '~/redux/hooks';

// Only exported for testing
// This is used ONLY if the api call fails to return a GA date
export const GA_DATE_STR = '2023-12-04T00:00:00Z';

export const createdPostGa = (creationDateStr: string, gaDateStr: string) => {
  const creationDate = creationDateStr ? new Date(creationDateStr) : new Date();
  const gaDate = gaDateStr ? new Date(gaDateStr) : new Date();
  return creationDate >= gaDate;
};

interface PreviewLabelProps extends LabelProps {
  creationDateStr: string;
  product?: string;
  type?: string;
}

export const PreviewLabel = (props: PreviewLabelProps) => {
  const { creationDateStr, product, type, ...rest } = props;
  const getPreview = useGetTechPreviewStatus(product || 'rosa', type || 'hcp');

  const techPreview = useTechPreviewStatus(product || 'rosa', type || 'hcp');

  React.useEffect(() => {
    if (!techPreview) {
      getPreview();
    }
  }, [getPreview, techPreview]);

  if (techPreview?.fulfilled) {
    const gaDate = techPreview?.end_date || GA_DATE_STR;
    return createdPostGa(creationDateStr, gaDate) ? null : (
      <Label
        color="green"
        icon={<InfoCircleIcon />}
        render={({ className, content }) => (
          <a
            className={className}
            href="https://access.redhat.com/support/offerings/techpreview"
            target="_blank"
            rel="noreferrer"
          >
            {content}
          </a>
        )}
        {...rest}
      >
        Preview
      </Label>
    );
  }
  return null;
};
