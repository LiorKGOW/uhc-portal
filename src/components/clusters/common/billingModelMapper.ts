/* eslint-disable camelcase */
import { RelatedResource } from '~/types/accounts_mgmt.v1';
import { BillingModel } from '~/types/clusters_mgmt.v1';

export const clusterBillingModelToRelatedResource = (
  clusterBillingModel?: BillingModel | string,
): RelatedResource.billing_model | undefined =>
  !['', undefined].includes(clusterBillingModel)
    ? (clusterBillingModel as unknown as RelatedResource.billing_model)
    : undefined;
