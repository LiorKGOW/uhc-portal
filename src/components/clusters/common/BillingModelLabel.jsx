import PropTypes from 'prop-types';

import get from 'lodash/get';

import { normalizedProducts, billingModels } from '../../../common/subscriptionTypes';

function BillingModelLabel({ cluster }) {
  const planId = get(cluster, 'subscription.plan.id');
  const billingModel = get(cluster, 'billing_model');
  const { ROSA, OSD, OSDTrial } = normalizedProducts;
  const { STANDARD, MARKETPLACE } = billingModels;
  const CCS = get(cluster, 'ccs.enabled');

  if (planId === ROSA) {
    return 'Through AWS';
  }

  if (planId === OSDTrial) {
    return 'Free trial, upgradeable';
  }

  // OSD non-ccs standard quota
  if (planId === OSD && billingModel === STANDARD) {
    return 'Subscription (yearly)';
  }

  // OSD CCS marketplace
  if (planId === OSD && billingModel === MARKETPLACE && CCS) {
    return 'Pay-as-you-go (hourly)';
  }

  return 'Standard';
}

BillingModelLabel.propTypes = {
  cluster: PropTypes.shape({
    product: PropTypes.shape({
      id: PropTypes.oneOf(Object.keys(normalizedProducts)).isRequired,
    }),
  }),
};

export default BillingModelLabel;
