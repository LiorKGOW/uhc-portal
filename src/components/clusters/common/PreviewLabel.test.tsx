import React from 'react';
import { mount } from 'enzyme';

import { PreviewLabel, GA_DATE, createdPostGa } from '~/components/clusters/common/PreviewLabel';

describe('PreviewLabel', () => {
  it('pre GA date', () => {
    const now = new Date(GA_DATE);
    now.setSeconds(GA_DATE.getSeconds() - 1);
    const wrapper = mount(<PreviewLabel creationDateStr={now.toISOString()} />);
    expect(wrapper).toMatchSnapshot();
    expect(createdPostGa(now.toISOString())).toBe(false);
  });

  it('post GA date', () => {
    const now = new Date(GA_DATE);
    const wrapper = mount(<PreviewLabel creationDateStr={now.toISOString()} />);
    expect(wrapper).toMatchSnapshot();
    expect(createdPostGa(now.toISOString())).toBe(true);
    now.setSeconds(GA_DATE.getSeconds() + 1);
    expect(wrapper).toMatchSnapshot();
    expect(createdPostGa(now.toISOString())).toBe(true);
  });
});
