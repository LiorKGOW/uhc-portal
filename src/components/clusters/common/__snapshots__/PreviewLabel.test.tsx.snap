// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewLabel post GA date 1`] = `
<PreviewLabel
  creationDateStr="2023-08-01T00:00:00.000Z"
/>
`;

exports[`PreviewLabel post GA date 2`] = `
<PreviewLabel
  creationDateStr="2023-08-01T00:00:00.000Z"
/>
`;

exports[`PreviewLabel pre GA date 1`] = `
<PreviewLabel
  creationDateStr="2023-07-31T23:59:59.000Z"
>
  <Label
    color="green"
    icon={
      <InfoCircleIcon
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />
    }
    render={[Function]}
  >
    <span
      className="pf-c-label pf-m-green"
    >
      <a
        className="pf-c-label__content"
        href="https://access.redhat.com/support/offerings/techpreview"
        rel="noreferrer"
        target="_blank"
      >
        <span
          className="pf-c-label__icon"
        >
          <InfoCircleIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              />
            </svg>
          </InfoCircleIcon>
        </span>
        Preview
      </a>
    </span>
  </Label>
</PreviewLabel>
`;
